### DISCUSSION TITLE
Segmentation fault(core dumped)

### DISCUSSION THREAD

**Discussion by Enjoy-Maity** [polybar/polybar#2653] [General]
I am facing this issue for a few days, well i am new to polybar and wanna try it out but whenever I try to launch the polybar i got to face this annoying statement:
#Segmentation fault (core dumped)
I don't know what's the problem here I haven't even tried to include any custom scripts all I did was made a few changes in the modules and then placed those modules  
Here's my --log=output

```
notice: Parsing config file: /home/enjoy/.config/polybar/config.ini
- config_parser: Parsing /home/enjoy/.config/polybar/config.ini
- config: Inheriting keys from "network-base" in "module/eth"
- config: Inheriting keys from "network-base" in "module/wlan"
* Loaded monitor eDP-1 (1366x768+0+0)
* Configured DPI = 96x96
* Bar geometry: 1366x24+0+0; Borders: 0,0,0,0
- bar: Attach X event sink
- bar: Attach signal receiver
- controller: Setup user-defined modules
Segmentation fault (core dumped)
```

someone please help me figure it out what's the problem

**Comment by patrick96** [comment]
Hi, thanks for reporting :)

Segmentation faults (segfaults) are *always* bugs in polybar itself, they are unfortunately also quite hard to debug from log messages alone.

Did this happen only after you made a few changes to your original config?

Please also post your entire config file and the output of the following command: `polybar -vvv`.

**Reply by Enjoy-Maity** [reply]
Here is my config file

**;==========================================================
;
;
; ██████╗ ██████╗ ██╗ ██╗ ██╗██████╗ █████╗ ██████╗
; ██╔══██╗██╔═══██╗██║ ╚██╗ ██╔╝██╔══██╗██╔══██╗██╔══██╗
; ██████╔╝██║ ██║██║ ╚████╔╝ ██████╔╝███████║██████╔╝
; ██╔═══╝ ██║ ██║██║ ╚██╔╝ ██╔══██╗██╔══██║██╔══██╗
; ██║ ╚██████╔╝███████╗██║ ██████╔╝██║ ██║██║ ██║
; ╚═╝ ╚═════╝ ╚══════╝╚═╝ ╚═════╝ ╚═╝ ╚═╝╚═╝ ╚═╝
;
;
; To learn more about how to configure Polybar
; go to https://github.com/polybar/polybar
;
; The README contains a lot of information
;
;==========================================================

[colors]
background = #282A2E
background-alt = #373B41
foreground = #C5C8C6
primary = #F0C674
secondary = #8ABEB7
alert = #A54242
disabled = #707880

[bar/example]
width = 100%
height = 24pt
radius = 6

; dpi = 96

background = ${colors.background}
foreground = ${colors.foreground}

line-size = 3pt

border-size = 4pt
border-color = #00000000

padding-left = 0
padding-right = 1

module-margin = 1

separator = |
separator-foreground = ${colors.disabled}

font-0 = monospace;2

modules-left = xworkspaces xwindow
modules-right = filesystem pulseaudio xkeyboard memory cpu wlan eth date

cursor-click = pointer
cursor-scroll = ns-resize

enable-ipc = true

; tray-position = right

;wm-restack = generic
; wm-restack = bspwm
; wm-restack = i3

; override-redirect = true

[module/xworkspaces]
type = internal/xworkspaces

label-active = %name%
label-active-background = ${colors.background-alt}
label-active-underline= ${colors.primary}
label-active-padding = 1

label-occupied = %name%
label-occupied-padding = 1

label-urgent = %name%
label-urgent-background = ${colors.alert}
label-urgent-padding = 1

label-empty = %name%
label-empty-foreground = ${colors.disabled}
label-empty-padding = 1

[module/xwindow]
type = internal/xwindow
label = %title:0:60:...%

[module/filesystem]
type = internal/fs
interval = 25

mount-0 = /

label-mounted = %{F#F0C674}%mountpoint%%{F-} %percentage_used%%

label-unmounted = %mountpoint% not mounted
label-unmounted-foreground = ${colors.disabled}

[module/pulseaudio]
type = internal/pulseaudio
sink = alsa_output.pci-0000_04_00.6.analog-stereo
use-ui-max = true
interval = 5

label-volume = %percentage%%
label-muted = 
;label-muted-background= https://github.com/polybar/polybar/issues/666[](https://github.com/Enjoy-Maity)
label-muted-foreground = ${colors.disabled}

ramp-volume-0 = 
ramp-volume-1 = 奔
ramp-volume-2 = 墳

;format-volume-prefix = "VOL "
format-volume-prefix-foreground = ${colors.primary}
format-volume =

click-right = pavucontrol

[module/xkeyboard]
type = internal/xkeyboard
blacklist-0 = num lock

label-layout = %layout%
label-layout-foreground = ${colors.primary}

label-indicator-padding = 2
label-indicator-margin = 1
label-indicator-foreground = ${colors.background}
label-indicator-background = ${colors.secondary}

[module/memory]
type = internal/memory
interval = 2
format-prefix = "RAM "
format-prefix-foreground = ${colors.primary}
label = %percentage_used:2%%

[module/cpu]
type = internal/cpu
interval = 2
format-prefix = "CPU "
format-prefix-foreground = ${colors.primary}
label = %percentage:2%%

[network-base]
type = internal/network
interval = 5
format-connected =
format-disconnected =
label-disconnected = %{F#F0C674}%ifname%%{F#707880} disconnected

[module/wlan]
inherit = network-base
interface-type = wireless
label-connected = %{F#F0C674}%ifname%%{F-} %essid% %local_ip%

[module/eth]
inherit = network-base
interface-type = wired
label-connected = %{F#F0C674}%ifname%%{F-} %local_ip%

[module/date]
type = internal/date
interval = 1

time = %I:%M %p
date = %d %h, %a
date-alt = %Y-%m-%d %H:%M:%S

label = %date% %time%
label-foreground = ${colors.primary}

[settings]
screenchange-reload = true
pseudo-transparency = true

[module/battery]
type = internal/battery
full-at=100
low-at=5
battery = BAT0
;adapter = AC
poll-interval=0

format-full=

ramp-capacity-0 = 
ramp-capacity-1 = 
ramp-capacity-2 = 
ramp-capacity-3 = 
ramp-capacity-4 = 
ramp-capacity-5 = 

animation-charging-0 = 
animation-charging-1 = 
animation-charging-2 = 
animation-charging-3 = 
animation-charging-4 = 

animation-charging-framerate = 750

format-charging =
format-discharging =

label-charging = Charging %percentage%%
label-discharging = Discharging %percentage%%

label-low=Battery Low
label-full = Fully charged

[bar/top]
bottom = false
;monitor = eDP1
background=${colors.background}
border-size =0
border-color= #000000

padding-right =2
module-margin-left=1
module-margin-right=1

font-0= Roboto Regular: size=10;1

modules-left= xworkspaces xwindow
modules-right = date wlan battery pulseaudio

; vim:ft=dosini**

and here is the output of polybar -vvv

polybar -vvv
polybar 3.6.1-20-g3079acb3

Features: -alsa -curl -i3 -mpd -network(libnl) -pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/g++
Compiler flags: -O3 -DNDEBUG -Wall -Wextra -Wpedantic -Wsuggest-override
Linker flags: -Wall -Wextra -Wpedantic -Wsuggest-override -Wall -Wextra -Wpedantic -Wsuggest-override
@Enjoy-Maity

**Comment by Enjoy-Maity** [comment]
Yes, I made certain changes like adding ramp-volume.... in pulseaudio
module , other than that I created a bar

**Reply by Enjoy-Maity** [reply]
Here is my config file

```
;==========================================================
;
;
;   ██████╗  ██████╗ ██╗  ██╗   ██╗██████╗  █████╗ ██████╗
;   ██╔══██╗██╔═══██╗██║  ╚██╗ ██╔╝██╔══██╗██╔══██╗██╔══██╗
;   ██████╔╝██║   ██║██║   ╚████╔╝ ██████╔╝███████║██████╔╝
;   ██╔═══╝ ██║   ██║██║    ╚██╔╝  ██╔══██╗██╔══██║██╔══██╗
;   ██║     ╚██████╔╝███████╗██║   ██████╔╝██║  ██║██║  ██║
;   ╚═╝      ╚═════╝ ╚══════╝╚═╝   ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝
;
;
;   To learn more about how to configure Polybar
;   go to https://github.com/polybar/polybar
;
;   The README contains a lot of information
;
;==========================================================

[colors]
background = #282A2E
background-alt = #373B41
foreground = #C5C8C6
primary = #F0C674
secondary = #8ABEB7
alert = #A54242
disabled = #707880

[bar/example]
width = 100%
height = 24pt
radius = 6

; dpi = 96

background = ${colors.background}
foreground = ${colors.foreground}

line-size = 3pt

border-size = 4pt
border-color = #00000000

padding-left = 0
padding-right = 1

module-margin = 1

separator = |
separator-foreground = ${colors.disabled}

font-0 = monospace;2

modules-left = xworkspaces xwindow
modules-right = filesystem pulseaudio xkeyboard memory cpu wlan eth date

cursor-click = pointer
cursor-scroll = ns-resize

enable-ipc = true

; tray-position = right

;wm-restack = generic
; wm-restack = bspwm
; wm-restack = i3

; override-redirect = true

[module/xworkspaces]
type = internal/xworkspaces

label-active = %name%
label-active-background = ${colors.background-alt}
label-active-underline= ${colors.primary}
label-active-padding = 1

label-occupied = %name%
label-occupied-padding = 1

label-urgent = %name%
label-urgent-background = ${colors.alert}
label-urgent-padding = 1

label-empty = %name%
label-empty-foreground = ${colors.disabled}
label-empty-padding = 1

[module/xwindow]
type = internal/xwindow
label = %title:0:60:...%

[module/filesystem]
type = internal/fs
interval = 25

mount-0 = /

label-mounted = %{F#F0C674}%mountpoint%%{F-} %percentage_used%%

label-unmounted = %mountpoint% not mounted
label-unmounted-foreground = ${colors.disabled}

[module/pulseaudio]
type = internal/pulseaudio
sink = alsa_output.pci-0000_04_00.6.analog-stereo
use-ui-max = true
interval = 5

label-volume = %percentage%%
label-muted =  
;label-muted-background= #666 
label-muted-foreground = ${colors.disabled}

ramp-volume-0 = 
ramp-volume-1 = 奔
ramp-volume-2 = 墳


;format-volume-prefix = "VOL "
format-volume-prefix-foreground = ${colors.primary}
format-volume = <ramp-volume> <label-volume>

click-right = pavucontrol

[module/xkeyboard]
type = internal/xkeyboard
blacklist-0 = num lock

label-layout = %layout%
label-layout-foreground = ${colors.primary}

label-indicator-padding = 2
label-indicator-margin = 1
label-indicator-foreground = ${colors.background}
label-indicator-background = ${colors.secondary}

[module/memory]
type = internal/memory
interval = 2
format-prefix = "RAM "
format-prefix-foreground = ${colors.primary}
label = %percentage_used:2%%

[module/cpu]
type = internal/cpu
interval = 2
format-prefix = "CPU "
format-prefix-foreground = ${colors.primary}
label = %percentage:2%%

[network-base]
type = internal/network
interval = 5
format-connected = <label-connected>
format-disconnected = <label-disconnected>
label-disconnected = %{F#F0C674}%ifname%%{F#707880} disconnected

[module/wlan]
inherit = network-base
interface-type = wireless
label-connected = %{F#F0C674}%ifname%%{F-} %essid% %local_ip%


[module/eth]
inherit = network-base
interface-type = wired
label-connected = %{F#F0C674}%ifname%%{F-} %local_ip%

[module/date]
type = internal/date
interval = 1

time = %I:%M %p
date = %d %h, %a
date-alt = %Y-%m-%d %H:%M:%S

label = %date% %time% 
label-foreground = ${colors.primary}

[settings]
screenchange-reload = true
pseudo-transparency = true

[module/battery]
type = internal/battery
full-at=100
low-at=5
battery = BAT0
;adapter = AC
poll-interval=0

format-full=<ramp-capacity> <label-full>


ramp-capacity-0 = 
ramp-capacity-1 = 
ramp-capacity-2 = 
ramp-capacity-3 = 
ramp-capacity-4 = 
ramp-capacity-5 = 

animation-charging-0 = 
animation-charging-1 = 
animation-charging-2 = 
animation-charging-3 = 
animation-charging-4 = 

animation-charging-framerate = 750


format-charging = <animation-charging> <label-charging>
format-discharging = <ramp-capacity> <label-discharging>

label-charging = Charging %percentage%%
label-discharging = Discharging %percentage%%

label-low=Battery Low
label-full = Fully charged

[bar/top]
bottom = false
;monitor = eDP1
background=${colors.background}
border-size =0
border-color= #000000

padding-right =2
module-margin-left=1
module-margin-right=1

font-0= Roboto Regular: size=10;1

modules-left= xworkspaces xwindow
modules-right = date wlan battery pulseaudio

; vim:ft=dosini
```





and here is the output of _polybar -vvv_


```
polybar -vvv
polybar 3.6.1-20-g3079acb3

Features: -alsa -curl -i3 -mpd -network(libnl) -pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/g++
Compiler flags:  -O3 -DNDEBUG -Wall -Wextra -Wpedantic -Wsuggest-override
Linker flags:  -Wall -Wextra -Wpedantic -Wsuggest-override  -Wall -Wextra -Wpedantic -Wsuggest-override
```

**Comment by patrick96** [comment]
I unfortunately can't reproduce this on my machine, so this will make debugging this a bit more difficult.
When the segfault occurs, you are running the `top` bar, correct?

In that case, it seems the error occurs while constructing either the `xworkspaces`, `xwindow`, `date`, or `wlan` module. The other two modules are likely not at fault because modules are initialized from left to right and the `wlan` module would have printed a log message if it successfully initialized, so the crash must have occured before that.

Try removing modules one-by-one from the `modules-left` and `modules-right` lists until the bar doesn't crash anymore.
This should narrow down which module is responsible for the crash and help us debug further.

**Reply by Enjoy-Maity** [reply]
Thanks , the problem was with the *xworkspaces* module, as soon as i removed it the polybar was working.

**Reply by patrick96** [reply]
That's good to know that there is a bug in the xworkspaces module, now we just need to find it ^^

What WM are you using?

**Reply by Enjoy-Maity** [reply]
I am using **Openbox**  floating window manager .

**Reply by patrick96** [reply]
I am currently trying to dig through the source code to find where this error could occur and I have so far only found one place where this could be happening. Do you mind posting the output of the command `xprop -root` to confirm my suspicion?

**Reply by Enjoy-Maity** [reply]
Here is the output of **xprop -root** :

```
xprop -root
_NET_ACTIVE_WINDOW(WINDOW): window id # 0x22002b0
PULSE_COOKIE(STRING) = "479397a1130c40d886c3773b89148e19dd390efec0ac95d61cde7f71367b25f00f7cb5bfa831a8a7181f8d6bcf598d5117efeeb1bdef5f1ecad0d745906d8eb43b11355fba9fbd60f18eb2854391e261850f3f636ac3d78f7cd6119593a14062740fef34d435f4bd8e7484f9f463a23d0455344922a6b51fcb319f355b5cf219884ac9698c8fe5316503f2605ededd0abb4cedd20d0eb629775629f1ecaff50b1e5d59e4dcb1a347f5367a2fd3a9ca353fe120c4e7f92432db31b1a54614629d5a1178899a2d651c8f714f3a47ed0bec21b5b209757da2dfa63927c05c4e10d2c5dda401364f69b64a2ad9abab23c66355cd71297ea348ccccfdf81d595e85d6"
PULSE_SERVER(STRING) = "{9752f5bf76ee434a983f3b79caafcb30}unix:/run/user/1000/pulse/native"
PULSE_ID(STRING) = "1000@9752f5bf76ee434a983f3b79caafcb30/715"
AT_SPI_BUS(STRING) = "unix:path=/tmp/dbus-e3k18e1WwL,guid=9ecbcd0172021cfe1caea1b26239ac03"
_NET_DESKTOP_NAMES(UTF8_STRING) = "desktop 1", "desktop 2", "desktop 3", "desktop 4"
_NET_CLIENT_LIST(WINDOW): window id # 0xc00003, 0x1e0002c, 0x2400007, 0x2600001, 0x22002b0, 0x3200007
_NET_SHOWING_DESKTOP(CARDINAL) = 0
_NET_WORKAREA(CARDINAL) = 0, 0, 1366, 768, 0, 0, 1366, 768, 0, 0, 1366, 768, 0, 0, 1366, 768
_NET_DESKTOP_VIEWPORT(CARDINAL) = 0, 0, 0, 0, 0, 0, 0, 0
_NET_DESKTOP_GEOMETRY(CARDINAL) = 1366, 768
_NET_CLIENT_LIST_STACKING(WINDOW): window id # 0x3200007, 0x2400007, 0x2600001, 0x1e0002c, 0x22002b0, 0xc00003
_OB_THEME(UTF8_STRING) = "Kaunas-1.0.1"
_OB_VERSION(UTF8_STRING) = "3.6.1"
_NET_SUPPORTED(ATOM) = _NET_SUPPORTING_WM_CHECK, _NET_WM_FULL_PLACEMENT, _NET_CURRENT_DESKTOP, _NET_NUMBER_OF_DESKTOPS, _NET_DESKTOP_GEOMETRY, _NET_DESKTOP_VIEWPORT, _NET_ACTIVE_WINDOW, _NET_WORKAREA, _NET_CLIENT_LIST, _NET_CLIENT_LIST_STACKING, _NET_DESKTOP_NAMES, _NET_CLOSE_WINDOW, _NET_DESKTOP_LAYOUT, _NET_SHOWING_DESKTOP, _NET_WM_NAME, _NET_WM_VISIBLE_NAME, _NET_WM_ICON_NAME, _NET_WM_VISIBLE_ICON_NAME, _NET_WM_DESKTOP, _NET_WM_STRUT, _NET_WM_STRUT_PARTIAL, _NET_WM_ICON, _NET_WM_ICON_GEOMETRY, _NET_WM_WINDOW_TYPE, _NET_WM_WINDOW_TYPE_DESKTOP, _NET_WM_WINDOW_TYPE_DOCK, _NET_WM_WINDOW_TYPE_TOOLBAR, _NET_WM_WINDOW_TYPE_MENU, _NET_WM_WINDOW_TYPE_UTILITY, _NET_WM_WINDOW_TYPE_SPLASH, _NET_WM_WINDOW_TYPE_DIALOG, _NET_WM_WINDOW_TYPE_NORMAL, _NET_WM_ALLOWED_ACTIONS, _NET_WM_WINDOW_OPACITY, _NET_WM_ACTION_MOVE, _NET_WM_ACTION_RESIZE, _NET_WM_ACTION_MINIMIZE, _NET_WM_ACTION_SHADE, _NET_WM_ACTION_MAXIMIZE_HORZ, _NET_WM_ACTION_MAXIMIZE_VERT, _NET_WM_ACTION_FULLSCREEN, _NET_WM_ACTION_CHANGE_DESKTOP, _NET_WM_ACTION_CLOSE, _NET_WM_ACTION_ABOVE, _NET_WM_ACTION_BELOW, _NET_WM_STATE, _NET_WM_STATE_MODAL, _NET_WM_STATE_MAXIMIZED_VERT, _NET_WM_STATE_MAXIMIZED_HORZ, _NET_WM_STATE_SHADED, _NET_WM_STATE_SKIP_TASKBAR, _NET_WM_STATE_SKIP_PAGER, _NET_WM_STATE_HIDDEN, _NET_WM_STATE_FULLSCREEN, _NET_WM_STATE_ABOVE, _NET_WM_STATE_BELOW, _NET_WM_STATE_DEMANDS_ATTENTION, _NET_MOVERESIZE_WINDOW, _NET_WM_MOVERESIZE, _NET_WM_USER_TIME, _NET_FRAME_EXTENTS, _NET_REQUEST_FRAME_EXTENTS, _NET_RESTACK_WINDOW, _NET_STARTUP_ID, _NET_WM_SYNC_REQUEST, _NET_WM_SYNC_REQUEST_COUNTER, _NET_WM_PID, _NET_WM_PING, _KDE_WM_CHANGE_STATE, _KDE_NET_WM_FRAME_STRUT, _KDE_NET_WM_WINDOW_TYPE_OVERRIDE, _OB_WM_ACTION_UNDECORATE, _OB_WM_STATE_UNDECORATED, _OPENBOX_PID, _OB_THEME, _OB_CONFIG_FILE, _OB_CONTROL, _OB_VERSION, _OB_APP_ROLE, _OB_APP_TITLE, _OB_APP_NAME, _OB_APP_CLASS, _OB_APP_GROUP_NAME, _OB_APP_GROUP_CLASS, _OB_APP_TYPE
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0x40008b
_OPENBOX_PID(CARDINAL) = 632
ESETROOT_PMAP_ID(PIXMAP): pixmap id # 0x600001
_XROOTPMAP_ID(PIXMAP): pixmap id # 0x600001
GDK_VISUALS(INTEGER) = 1319, 1737
_XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us", "", ""
XFree86_has_VT(INTEGER) = 1
XFree86_VT(INTEGER) = 7
Xorg_Seat(STRING) = "seat0"
```

**Reply by patrick96** [reply]
Thanks. It looks like openbox is not setting `_NET_NUMBER_OF_DESKTOPS` which may be tripping up the module. I'll look into it

**Comment by Enjoy-Maity** [comment]
Thanks!

**Reply by patrick96** [reply]
Do you remove the `_NET_NUMBER_OF_DESKTOPS` atom (with a command like `xprop -root -remove _NET_NUMBER_OF_DESKTOPS`) when openbox starts? Because normally openbox does set `_NET_NUMBER_OF_DESKTOPS`.

**Reply by Enjoy-Maity** [reply]
No I didn't do anything, I never used xprop so how can  I remove _NET_NUMBER_OF_DESKTOPS
via xprop 

I think it's something to do with the OS I am using, I am using **Arch Linux** with **openbox** window manager

**Reply by patrick96** [reply]
I have been trying to reproduce this, also on Arch, and I was unable to get openbox to *not* set `_NET_NUMBER_OF_DESKTOPS`. There is something in your setup either preventing it from being set or deleting it.

This just means that your setup will not be able to use the xworkspaces module (even after this crash bug is fixed) because without `_NET_NUMBER_OF_DESKTOPS`, the module will just assume there are no workspaces.

**Reply by Enjoy-Maity** [reply]
Okay, But can you give me some advice on how to activate or set the openbox to get 
_NET_NUMBER_OF_DESKTOPS

**Reply by patrick96** [reply]
I have no idea. I couldn't even get openbox to *not* set it, even if I completely removed the setting from its config file. People in openbox support channels may have more insight here.
